src/Command/Site/BuildCommand.php

```diff
@@ -0,0 +1,317 @@
+   */
+  private $output;
+
+  /**
+   * Stores global options passed.
+   */
+  private $inputOptions;
+
+  /**
+   * Stores the commands to be skipped.
+   */
+  private $skip;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function configure() {
+    parent::configure();
+
+    $commands = array(
+      'site:checkout',
+      'site:compose|make',
+      'site:npm',
+      'site:grunt',
+      'site:settings',
+      'site:phpunit:setup',
+      'site:behat:setup',
+      'site:db:import',
+      'site:update',
+    );
+    $this->setName('site:build')
+      ->setDescription(sprintf('Runs the following commands to build a site: %s.', implode(', ', $commands)));
+
+    // Custom options.
+    $this->addOption(
+      'branch',
+      '-B',
+      InputOption::VALUE_OPTIONAL,
+      'Specify which branch to build if different than the global branch found in sites.yml'
+    );
+    $this->addOption(
+      'skip',
+      '',
+      InputOption::VALUE_OPTIONAL,
+      'Used to skip one or more commands. i.e. --skip="checkout, phpunit:setup"'
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function interact(InputInterface $input, OutputInterface $output) {
+    parent::interact($input, $output);
+
+    $this->input = $input;
+    $this->output = $output;
+    $this->inputOptions = array_filter($input->getOptions());
+
+    $this->skip = array();
+    if (isset($this->inputOptions['skip'])) {
+      $this->skip = explode(',', $this->inputOptions['skip']);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function execute(InputInterface $input, OutputInterface $output) {
+    parent::execute($input, $output);
+
+    $this->commands = array();
+
+    $this->addCheckoutCommand();
+    // Run checkout first because Compose/Make depends on it.
+    $this->runList();
+
+    $this->addComposeMakeCommand();
+    $this->addNPMCommand();
+    $this->addGruntCommand();
+    $this->addSettingsCommand();
+    $this->addTestSetupCommand();
+    $this->addDbImportCommand();
+    $this->addUpdateCommand();
+    $this->runList();
+  }
+
+  /**
+   * Runs the commands.
+   */
+  private function runList() {
+    foreach ($this->commands as $item) {
+      $parameters = array();
+      $command = $this->getApplication()->find($item['command']);
+
+      // Command arguments.
+      if (!empty($item['arguments'])) {
+        foreach ($item['arguments'] as $name => $value) {
+          $parameters[$name] = $value;
+        }
+      }
+
+      // Command options.
+      if (isset($item['options'])) {
+        $options = array_filter($item['options']);
+        foreach ($options as $name => $value) {
+          $parameters['--' . $name] = $value;
+        }
+      }
+
+      // Append env if needed.
+      if (isset($this->inputOptions['env'])) {
+        $parameters['--env'] = $this->inputOptions['env'];
+      }
+
+      $this->io->writeln(sprintf('// %s', $item['command']));
+      //echo var_export($parameters, true) . PHP_EOL;
```

Comment:
Small thing: does this need to be removed?

Reply:
The commented line? yes